File: OBLIVIOUS-DEV/project/legacy-frozen/AiMindBridge/AiMindBridge.Server/Providers/QwenAdapter.cs
using System.Text;
using System.Text.Json;
using AiMindBridge.Config;
using Microsoft.Extensions.Logging;

namespace AiMindBridge.Providers;

/// <summary>Alibaba Qwen adapter — uses DashScope-compatible endpoint.</summary>
public sealed class QwenAdapter : ProviderAdapterBase
{
    public override string Name => "qwen";
    private static readonly string DefaultModel = "qwen-turbo";
    private static readonly string BaseUrl       = "https://dashscope.aliyuncs.com/compatible-mode/v1/chat/completions";

    public QwenAdapter(ProviderSettings settings, ILogger<QwenAdapter> logger)
        : base(settings, logger) { }

    protected override object BuildHttpPayload(string prompt, string apiKey)
    {
        Http.DefaultRequestHeaders.Clear();
        Http.DefaultRequestHeaders.Add("Authorization", $"Bearer {apiKey}");
        return new
        {
            model       = string.IsNullOrEmpty(Settings.Model) ? DefaultModel : Settings.Model,
            max_tokens  = Settings.MaxTokens,
            temperature = Settings.Temperature,
            messages    = new[]
            {
                new { role = "system", content = "You are a professional forex trade planner. Always respond only with valid JSON." },
                new { role = "user",   content = prompt }
            }
        };
    }

    protected override async Task<string> SendHttpAsync(object payload, CancellationToken ct)
    {
        var url     = string.IsNullOrEmpty(Settings.BaseUrl) ? BaseUrl : Settings.BaseUrl;
        var json    = JsonSerializer.Serialize(payload);
        var content = new StringContent(json, Encoding.UTF8, "application/json");
        var resp    = await Http.PostAsync(url, content, ct);
        resp.EnsureSuccessStatusCode();
        var body = await resp.Content.ReadAsStringAsync(ct);
        using var doc = JsonDocument.Parse(body);
        return doc.RootElement
            .GetProperty("choices")[0]
            .GetProperty("message")
            .GetProperty("content")
            .GetString() ?? "";
    }
}

File: oblivious-hub/src/services/providers/QwenAdapter.js
// Qwen (Alibaba DashScope OpenAI-compatible endpoint).
const { BaseAdapter, httpsRequest } = require("./_BaseAdapter");

class QwenAdapter extends BaseAdapter {
  constructor({ apiKey }) { super({ apiKey, name: "qwen" }); }

  async _doQuery({ system, user }) {
    return httpsRequest({
      url:    "https://dashscope-intl.aliyuncs.com/compatible-mode/v1/chat/completions",
      method: "POST",
      headers: {
        "Content-Type":  "application/json",
        "Authorization": `Bearer ${this.apiKey}`,
      },
      body: {
        model: "qwen-turbo",
        messages: [
          { role: "system", content: system || "" },
          { role: "user",   content: user   || "ping" },
        ],
        max_tokens:  40,
        temperature: 0,
      },
      timeoutMs: 10_000,
    });
  }
  _extractContent(json) {
    return json?.choices?.[0]?.message?.content || JSON.stringify(json || {}).slice(0, 600);
  }

  // DashScope compat-mode exposes the standard OpenAI GET /v1/models route.
  // Zero tokens consumed — works even on free-tier keys without quota.
  async _doTest() {
    return httpsRequest({
      url:    "https://dashscope-intl.aliyuncs.com/compatible-mode/v1/models",
      method: "GET",
      headers: { "Authorization": `Bearer ${this.apiKey}` },
      timeoutMs: 8_000,
    });
  }
}

module.exports = { QwenAdapter };
